src/server.c

```diff
@@ -78,6 +79,81 @@ drop_permissions(void)
 	}
 }
 
+static void seat_clear_touch_points(struct seat *seat,
+				    struct wl_client *active_client){
+	struct timespec now;
+	clock_gettime(CLOCK_MONOTONIC, &now);
+	struct wlr_touch_point *point;
+	wl_list_for_each(point, &seat->seat->touch_state.touch_points, link) {
+		if (point->client->client != active_client) {
+			wlr_seat_touch_point_clear_focus(seat->seat,
+							 now.tv_nsec / 1000, point->touch_id);
+		}
+	}
+}
+
+
+static void seat_inhibit_input(struct seat *seat,  struct wl_client *active_client)
+{
+	seat->active_client_while_inhibited = active_client;
+
+	if(seat->focused_layer &&
+	   (wl_resource_get_client(seat->focused_layer->resource) !=
+	    active_client))
+	{
+		wlr_log(WLR_INFO, "defocus layer");
+		seat_set_focus_layer(seat, NULL); /* ? */
```

Comment:
I note your question mark. I think is fine.
If the active_client is in a layer, and that layer is the focused one, then leave it - otherwise set focused layer to NULL

Reply:
Hah, good spot. I put the comments in to keep straight in my head the difference between pointer focus and keyboard focus, will delete